data/transactions/logic/eval.go

```diff
@@ -5700,17 +5700,22 @@ func opBlock(cx *EvalContext) error {
 	switch fs.field {
 	case BlkSeed:
 		cx.Stack[last].Bytes = hdr.Seed[:]
-		return nil
 	case BlkTimestamp:
 		cx.Stack[last].Bytes = nil
 		if hdr.TimeStamp < 0 {
 			return fmt.Errorf("block(%d) timestamp %d < 0", round, hdr.TimeStamp)
 		}
 		cx.Stack[last].Uint = uint64(hdr.TimeStamp)
-		return nil
+	case BlkProposer:
+		cx.Stack[last].Bytes = hdr.Proposer[:]
+	case BlkFeesCollected:
+		cx.Stack[last].Uint = hdr.FeesCollected.Raw
```

Comment:
It's safe, but almost by coincidence, so I'll update all the uint return values to assign the entire stackValue.
(It's safe because the stack value being written to is already a uint64, so it `.Bytes` is already `nil`)

Reply:
https://github.com/algorand/go-algorand/pull/5757/commits/a7e701063e99563dd7d4e26fa8fe24e345d54220

Reply:
We do have a test that would have failed if the "coincidence" didn't make this work.
`func TestBlockFieldsVersions(t *testing.T)`
https://github.com/jannotti/go-algorand/blob/absenteeism/data/transactions/logic/fields_test.go#L312